src/plugins/embeddable/public/embeddable_panel/embeddable_panel.tsx

```diff
@@ -129,28 +129,42 @@ export const EmbeddablePanel = (panelProps: UnwrappedEmbeddablePanelProps) => {
    * Select state from the embeddable
    */
   const loading = useSelectFromEmbeddableOutput('loading', embeddable);
+
+  if (loading === false && !initialLoadComplete) {
+    setInitialLoadComplete(true);
+  }
```

Comment:
I suspect this is relying on the fact that possible values for this state atom boolean, and multiple set of the same value doesn't trigger a re-render.
But in general call a `setState` within a render` function is not a good idea.
So I would propose to wrap this into a `useEffect`:

```ts
useEffect(() => {
    if (loading === false && !initialLoadComplete) {
      setInitialLoadComplete(true);
    }
  }, [initialLoadComplete, loading]);
```

Reply:
Am I correct in understanding that this will introduce another render?

> But in general call a setState within a render function is not a good idea.

Why is this? The only reason I can think of is the possibility of triggering a render loop, but you can do that with a `useEffect` too.